shell/ev-application.c

```diff
@@ -419,9 +419,18 @@ on_register_uri_cb (GObject      *source_object,
 	g_variant_builder_add (&builder, "{sv}",
 	                       "display",
 	                       g_variant_new_string (gdk_display_get_name (gdk_screen_get_display (data->screen))));
-	g_variant_builder_add (&builder, "{sv}",
-	                       "screen",
-	                       g_variant_new_int32 (gdk_x11_screen_get_screen_number (data->screen)));
+
+	if (GDK_IS_X11_DISPLAY (gdk_display_get_default())){
```

Comment:
Better check the actual screen that will be used:

```suggestion
	if (GDK_IS_X11_SCREEN (data->screen)) {
```

Reply:
Just changed this is a force-push, thanks